libportal/meson.build

```diff
@@ -109,12 +109,17 @@ if introspection
   endif
 endif
 
+enabled_backends = []
+
 ########
 # GTK3 #
 ########
 
-if 'gtk3' in backends
-  gtk3_dep = dependency('gtk+-3.0')
+require_gtk3 = 'gtk3' in backends
+want_gtk3 = 'auto' in backends or require_gtk3
+
+gtk3_dep = dependency('gtk+-3.0', required: require_gtk3)
+if want_gtk3 and gtk3_dep.found()
```

Comment:
This will do a lookup for gtk3 even if the value is `qt5` and not auto. So, perhaps:

```meson
gtk3_dep = dependency('', required: false)

if want_gtk3
    gtk3_dep = dependency('gtk+-3.0', required: require_gtk3)
endif
if gtk3_dep.found()
    ...
```

Reply:
does the lookup matter much? Personally I'm in the camp of simpler code over lookups but if it comes at a cost other than the print line and a tiny amount of time, I can change this.

Reply:
Well, even the time can be non-negligible considering that if gtk3 is not installed, meson will look up cmake, then run a cmake build in order to check if cmake can find it using a `find_package()` that will also eventually fail.

This can be confusing to users, apparently, due to mentioning cmake. ;) Why buy the trouble for a situation you know you will ignore it?

On the other hand you can fix that with a different code complexity. Specify the `method: ` kwarg.

...

Really, the general solution is feature options. Changing the array option into three different feature options may not be 100% desirable though... there have been discussions about instantiating custom feature objects, something like `make_feature_option('gtk3', required: require_gtk3, auto: want_gtk3)` except with a much better name and API obviously.

That's the magic I wish I could suggest.

Reply:
Changed as suggested!